### ISSUE TITLE
Wrong chart when the server timezone is UTC

### ISSUE THREAD

**Issue by anki-code** [metabase/metabase#4245]
![image](https://cloud.githubusercontent.com/assets/1708680/22327761/dd893c58-e3c9-11e6-93d8-457c63b553a5.png)

![image](https://cloud.githubusercontent.com/assets/1708680/22327785/fec698a2-e3c9-11e6-9f92-ae503a7d1df4.png)

Data for reproduce the bug:
```
select '2016-11-01T00:00:00.000Z'::datetime as dt,'A' as ctg, 12 as nm union all
select '2016-11-01T00:00:00.000Z'::datetime,'B', 39 union all
select '2016-12-01T00:00:00.000Z'::datetime,'C', 99 union all
select '2016-12-01T00:00:00.000Z'::datetime,'A', 164 union all
select '2016-12-01T00:00:00.000Z'::datetime,'B', 154 union all
select '2017-01-01T00:00:00.000Z'::datetime,'A', 94 union all
select '2017-01-01T00:00:00.000Z'::datetime,'B', 199

```

⏬ Please click the like icon instead of "+1" messages.

**Comment by anki-code** [comment]
All charts are affected (dates on scale and on line are shifted):
![image](https://cloud.githubusercontent.com/assets/1708680/22328094/7dd068b6-e3cb-11e6-9554-15a8a01a9b37.png)

**Comment by tlrobinson** [comment]
@anki-code I'm not able to reproduce this without some more information:

1. what database are you using?
2. what is your "Report Timezone" setting set to?
3. what timezone is your local compute / browser in?
4. what timezone is the machine running Metabase in?
5. what timezone is the machine running your database in?

**Comment by anki-code** [comment]
1. Vertica
2. Not set ( Defaults to system timezone.)
3. UTC+3
4. UTC
5. UTC

I'll recheck timezones but my configuration is stable. I've updated only Metabase.

**Comment by tlrobinson** [comment]
@anki-code Can you try the same query against one of our supported databases? The `Sample Dataset` (H2) would be fine for now. If you could also tell me the results of the `POST /api/dataset` requests for both Vertica and H2 (or another db) that would be great.

**Comment by anki-code** [comment]
I can reproduce the bug on "Sample Dataset" too and I see in Chrome console:
![image](https://cloud.githubusercontent.com/assets/1708680/22344246/730d5f1a-e40c-11e6-83b9-e38b282fadd1.png)

**Comment by anki-code** [comment]
@tlrobinson also I compare two versions:
1. Fresh Metabase 0.22.1 on my Windows Desktop (system timezone GMT+3; set Metabsase timezone to UTC)
2. Metabase 0.22.1 on my server (system timezone UTC; set Metabsase timezone to UTC)

On my desktop the bug has not reproduced on "Sample Dataset". But reproduced on the server.

`/api/dataset` on my desktop:
`{"finished_at":"2017-01-26T18:21:53.555Z","started_at":"2017-01-26T18:21:53.487Z","json_query":{"database":1,"type":"native","native":{"query":"select '2016-11-01T00:00:00.000Z'::datetime as dt,'A' as ctg, 12 as nm union all\r\nselect '2016-11-01T00:00:00.000Z'::datetime,'B', 39 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'C', 99 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'A', 164 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'B', 154 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'A', 94 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'B', 199"},"parameters":[],"constraints":{"max-results":10000,"max-results-bare-rows":2000}},"status":"completed","id":28,"uuid":"0f791340-ac60-4393-aa7a-5e6c11b0fc36","row_count":7,"running_time":68,"data":{"rows":[["2016-11-01T03:00:00.000+03:00","A",12],["2016-11-01T03:00:00.000+03:00","B",39],["2016-12-01T03:00:00.000+03:00","C",99],["2016-12-01T03:00:00.000+03:00","A",164],["2016-12-01T03:00:00.000+03:00","B",154],["2017-01-01T03:00:00.000+03:00","A",94],["2017-01-01T03:00:00.000+03:00","B",199]],"columns":["DT","CTG","NM"],"cols":[{"name":"DT","base_type":"type/DateTime"},{"name":"CTG","base_type":"type/Text"},{"name":"NM","base_type":"type/Integer"}],"native_form":{"query":"select '2016-11-01T00:00:00.000Z'::datetime as dt,'A' as ctg, 12 as nm union all\r\nselect '2016-11-01T00:00:00.000Z'::datetime,'B', 39 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'C', 99 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'A', 164 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'B', 154 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'A', 94 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'B', 199"}}}`

`/api/dataset` on the server:
`{"finished_at":"2017-01-26T18:20:00.299Z","started_at":"2017-01-26T18:20:00.273Z","json_query":{"database":5,"type":"native","native":{"query":"select '2016-11-01T00:00:00.000Z'::datetime as dt,'A' as ctg, 12 as nm union all\r\nselect '2016-11-01T00:00:00.000Z'::datetime,'B', 39 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'C', 99 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'A', 164 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'B', 154 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'A', 94 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'B', 199"},"parameters":[],"constraints":{"max-results":10000,"max-results-bare-rows":2000}},"status":"completed","id":12400,"uuid":"8cbc9d18-a0f6-455a-a441-6f6c8b8eaf49","row_count":7,"running_time":26,"data":{"rows":[["2016-11-01T00:00:00.000Z","A",12],["2016-11-01T00:00:00.000Z","B",39],["2016-12-01T00:00:00.000Z","C",99],["2016-12-01T00:00:00.000Z","A",164],["2016-12-01T00:00:00.000Z","B",154],["2017-01-01T00:00:00.000Z","A",94],["2017-01-01T00:00:00.000Z","B",199]],"columns":["DT","CTG","NM"],"cols":[{"name":"DT","base_type":"type/DateTime"},{"name":"CTG","base_type":"type/Text"},{"name":"NM","base_type":"type/Integer"}],"native_form":{"query":"select '2016-11-01T00:00:00.000Z'::datetime as dt,'A' as ctg, 12 as nm union all\r\nselect '2016-11-01T00:00:00.000Z'::datetime,'B', 39 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'C', 99 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'A', 164 union all\r\nselect '2016-12-01T00:00:00.000Z'::datetime,'B', 154 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'A', 94 union all\r\nselect '2017-01-01T00:00:00.000Z'::datetime,'B', 199"}}}
`

**Comment by anki-code** [comment]
Timezone from logs on server:
```
Starting Metabase version v0.22.1 (5fd5a44 release-0.22.1)
System timezone is 'Etc/UTC'
```

**Comment by anki-code** [comment]
@tlrobinson how to reproduce the bug:

1. Try it on Ubuntu server
2. Change timezone to UTC:
`dpkg-reconfigure tzdata`
3. Try SQL query which posted above and get the bug

If I change timezone to another GMT+n or GMT-n everything fine.

**Comment by anki-code** [comment]
@tlrobinson please fix it as P1 😿

**Comment by anki-code** [comment]
@salsakran could you please fix this bug to 0.22.2?

**Comment by salsakran** [comment]
The other bug we shipped a fix for in 0.22.2 broke startup entirely and had to go out asap. 

I'm trying to reproduce this.

**Comment by anki-code** [comment]
@salsakran @tlrobinson  guys, any news? This is a regress and I have a broken charts with monthly statistics:

![image](https://cloud.githubusercontent.com/assets/1708680/22364475/e144fbf2-e482-11e6-9f01-091d9dc5fd2c.png)

Can you suggest any workaround solution for this bug?

**Comment by salsakran** [comment]
@anki-code yes, there were no migrations in 0.22.1, so you should be ok to downgrade.

**Comment by anki-code** [comment]
@salsakran can you reproduce the bug? Can I help you?

**Comment by tlrobinson** [comment]
@anki-code Have you had a chance to try downgrading? I don't believe this is a regression. I was able to reproduce the issue in 0.22.1, 0.22.0, and 0.22.1.

**Comment by anki-code** [comment]
@tlrobinson you're right. I've downgraded Metabase to 0.21 and the bug is reproduced. I've understood why it affects me only now. I've changed SQL queries for monthly requests and catch the bug. Sorry for confusing. It's not a regress. The bug was before.

**Comment by anki-code** [comment]
@tlrobinson will you plan to fix this in next release?

Could you please set an actual labels for this issue (Unable to reproduce → Bug)?

**Comment by anki-code** [comment]
Guys, will be any workaround or fix this issue in 0.23?

**Comment by neoecos** [comment]
I had the same issue +1

**Comment by anki-code** [comment]
@camsaul what help you need?

**Comment by mazameli** [comment]
This appears to be fixed by #5729. I'm going to close this issue, but if those of you who've commented on this issue could [test out the release candidate we have for version 0.26](https://github.com/metabase/metabase/releases/tag/v0.26.0.RC1) which includes the fix, that would be much appreciated.

**Comment by mazameli** [comment]
In looking at the screenshots again more carefully, it looks like it also might be possible that @anki-code specifically might be experiencing the [issue where we're incorrectly appending a default time](https://github.com/metabase/metabase/issues/5859) to date fields which don't have an explicit time value. 

I'm going to reopen this until we're sure.

**Comment by stefanproell** [comment]
We are facing the same issue where the x-Axis of the graph is shifted by one day, giving us the last day of the previous month (2016-12-31) instead of the first (2017-01-01). 

Our metabase instance runs inside a Docker container and talks to a MySQL 5.6 database (inside a different Docker container). Metabase query timezone is set to UTC. Host, database and containers are all in UTC:

```
09-04 08:56:51 INFO metabase.core :: Starting Metabase version v0.25.2 (35bce2d release-0.25.2) ...
09-04 08:56:51 INFO metabase.core :: System timezone is 'UTC' ...
```

```
MySQL [(none)]> show variables like '%time_zone';
+------------------+--------+
| Variable_name    | Value  |
+------------------+--------+
| system_time_zone | UTC    |
| time_zone        | +00:00 |
+------------------+--------+
2 rows in set (0.00 sec)
```

We aggregate data and sum by month from one table in the MySQL instance. We discovered that when the date column uses a numeric format, but does not include the whole datetime string, then the problem occurs and the first month gets shifted leading to a wrong graph, exaclty as @anki-code showed before. 

We get the month by reformatting the data like this: 

`date_format(example.record_date, '%Y-%M') as monthly_aggregate`

or this 

`str_to_date(concat(date_format(example.record_date, '%Y-%m'), '-01'), '%Y-%m-%d') as monthly_aggregate`

which both produce the erroneous graph. However, if we use a date format that is non-numeric, we get the correct graph. So as a quick fix we can recommend to rewrite the query like this: 

`date_format(example.record_date, '%Y-%M') as monthly_aggregate,`

Notice the capitalized M which gives the full month name (e.g. 2017-April). With this modification, the graph displays the correct date.

**Comment by anki-code** [comment]
JFYI. The issue is actual for 0.26.1 version.

**Comment by anki-code** [comment]
Sad but still actual for 0.27.1

**Comment by TrineFN** [comment]
Also actual for 0.31.2

**Comment by mazameli** [comment]
I'm pretty sure this is fixed in master if folks want to try this out. I set my browser's timezone to something different from my reporting timezone, and was able to repro the problem in 0.32.10

## 0.32.10

![Screen Shot 2019-11-27 at 3 57 26 PM](https://user-images.githubusercontent.com/2223916/69766845-b620eb80-112e-11ea-8023-5334b680b2cd.png)

## master (upcoming 0.34)

![Screen Shot 2019-11-27 at 3 52 56 PM](https://user-images.githubusercontent.com/2223916/69766878-cdf86f80-112e-11ea-9bc0-6aa3b80f6cd4.png)

**Comment by mazameli** [comment]
Correction — this actually appears to be already fixed in 0.33.6. I'm guessing due to #11111.
